### ISSUE TITLE
parse() incorrect when len_size is greater than 1

### ISSUE THREAD

**Issue by laodengcode** [ustropo/uttlv#16]
When I tried to use `parse()` to parse a TLV with `len_size==2`, it cannot parse correctly. Reproduce:

```
t = TLV(len_size=2, tag_size=1)
map = {0x0b:{TLV.Config.Type:bytes}}
t.set_local_tag_map(map)

bin = b"\x0b\x02w-----BEGIN CERTIFICATE-----\nMIIBpjCCAUygAwIBAgIBATAKBggqhkjOPQQDAjBgMQswCQYDVQQGEwJERTEPMA0G\nA1UECAwGQkFZRVJOMQ8wDQYDVQQHDAZNVU5JQ0gxFDASBgNVBAoMC1N3aXNzYml0\nIEFHMRkwFwYDVQQDDBBTd2lzc2JpdCByb290IENBMB4XDTIzMDgxNDIwMDYwM1oX\nDTMzMDgxMDIwMDYwM1owWTELMAkGA1UEBhMCREUxDzANBgNVBAgMBkJBWUVSTjEP\nMA0GA1UEBwwGTVVOSUNIMRQwEgYDVQQKDAtTd2lzc2JpdCBBRzESMBAGA1UEAwwJ\nTXkgU2VydmVyMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEw76/1na5KrKcjvf7\nRqIVcOP8bPR86J9dOF7RaYV1GWwiBzrOyjKCPRSfQ+MVRqnL5rKEj12HvCOKWpcU\ni1+O1jAKBggqhkjOPQQDAgNIADBFAiBjrNQK4gm5gULqk0HYuP8JgI8kFCTwBRYX\nY+YCF9IGrAIhALf7SMSSjH8KnxSOaDLkthkLPk/6hx2ik/EIV97YYNJm\n-----END CERTIFICATE-----\n"

t.parse_array(bin)
print(t.tree())
```

The problem I found is because this [`offset`](https://github.com/ustropo/uttlv/blob/d53d5346ca7e1dc01bc1a6ca5eab6a0b86481e0b/uttlv/tlv.py#L247C47-L247C47) reduce the 2-bytes-length to 1

The `offset` should always be 0, right?

**Comment by Greenvi4** [comment]
I have the same promble. If offset set to zero it's works fine

```python
len_size = self.len_size or self.decode_len_size(aux)
# offset = 0 if len_size == 1 else 1
length = int.from_bytes(aux[:len_size], byteorder=self.endian)
```

It you could see, if use little endian

```python
tag_little = TLV(len_size=2, endian='little')
arr = [0x01, 0x04, 0x00, 0x10, 0x00, 0x00, 0x00]
tag_little.parse_array(arr)

assert tag_little[0x01] == 16
```

But I found another one bug: `IntEncoder` using only big endian!

```python
class IntEncoder(DefaultEncoder):
    def default(self, obj):
        if isinstance(obj, int):
            return obj.to_bytes(4, byteorder="big")
        return super().default(obj)

    def parse(self, obj, _cls):
        return int.from_bytes(obj, byteorder="big")
```

**Comment by dudanov** [comment]
Same problem. Thanks for the fix in #19 PR, @Greenvi4

**Comment by ustropo** [comment]
Fix in PR #19
